src/ralph/virtual/models.py

```diff
@@ -44,6 +46,13 @@ class Meta:
         null=False, blank=False, default=False
     )
     cloud_sync_driver = models.CharField(max_length=128, null=True, blank=True)
+    client_config = encrypt(
+        JSONField(
+            blank=True,
+            null=True,
+            verbose_name='client configuration',
```

Comment:
how the user should know what should he type into this field? text, json? if json, what keys?

Reply:
The validator ensures an input is always a valid JSON. I would not force any restrictions for the content because the latter depends on the needs for every specific client. I think it's better to add the description of the configuration for every specific client to the documentation.

Reply:
Alternatively, it's possible to use YAML instead of JSON. However, that would be more error-prone due to YAML being dependent on whitespaces.

Reply:
Could you add some example and/or config for specific clients to help_text and/or docs?

Reply:
Will do

Reply:
@romcheg how is it going with docs?